### PR TITLE
feat(website): automatically publish latest blog posts to website

### PR DESCRIPTION
**Pull request by Jaapiehetaapie**
#12277

### CODE REVIEW

**Comment by wolfderechter** [comment]
image
In darkmode the author name should be pink.

I don't think the authors dictionary is such a good idea..

File: packages/website/components/BlogSection.tsx [hunk 1]
```diff
@@ -1,61 +1,119 @@
...
+    <div className="relative bg-neutral-50 px-4 pt-16 pb-20 sm:px-6 lg:px-8 lg:pt-24 lg:pb-28 dark:bg-neutral-800">
       <div className="absolute inset-0">
-        <div className="h-1/3 bg-white sm:h-2/3 dark:bg-neutral-900" />
+        <div className="h-1/3 bg-white sm:h-2/3 dark:bg-[#1B1B1D]" />
       </div>
       <div className="relative mx-auto max-w-7xl">
         <div className="text-center">
           <h2 className="font-oxanium text-3xl font-bold tracking-tight text-neutral-900 sm:text-4xl dark:text-neutral-100">
-            Latest blog posts
+            Latest Blog Posts
```
Comment (wolfderechter) [inline packages/website/components/BlogSection.tsx:116]:
```suggestion
            Latest blog posts
```

File: packages/website/components/BlogSection.tsx [hunk 2]
```diff
@@ -67,46 +125,61 @@ export default function BlogSection() {
...
+                      <a>{post.authorship.name}</a>
                     </div>
                     <div className="flex space-x-1 text-sm text-neutral-500 dark:text-neutral-400">
-                      <time dateTime={post.datetime}>{post.date}</time>
+                      <time dateTime={getDateTime(post.content.timestamp)}>
+                        {getDate(post.content.timestamp)}
+                      </time>
                       <span aria-hidden="true">&middot;</span>
-                      <span>{post.readingTime} read</span>
+                      <span>{getReadingTime(post.content.body) + " min"}</span>
```
Comment (wolfderechter) [inline packages/website/components/BlogSection.tsx:182]:
```suggestion
                      <span>{getReadingTime(post.content.body) + " min read"}</span>
```

**Review by wolfderechter — CHANGES_REQUESTED** [review]

**Review by dantaik — DISMISSED** [review]

**Comment by dantaik** [comment]
@Jaapiehetaapie thank you for working on this. Could you please resolve the conflits?

**Comment by dantaik** [comment]
> The author name should be pink.
> I don't think the authors dictionary is such a good idea..

Lets remove the author's name and avatar.

**Comment by Jaapiehetaapie** [comment]
The other dependicies are needed for my code. Is their a way to resolve these confilcts?

File: packages/website/components/getPosts.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                  }
+              `})
+    }).then((res) => res.json())
+      .then((response) => {
+        const transactionCount = response.data.transactions.edges.length;
+
+        getPosts(response, transactionCount)
+      })
+      .catch((error) => {
+        console.log("An error occurred: ", error);
```
Comment (wolfderechter) [inline packages/website/components/getPosts.ts:48]: I'd remove all the console.logs

File: packages/website/components/getImgURLs.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                    if (!cardImgURL.includes("data")) {
+                        const urlStartIndex = cardImgURL.indexOf("url=") + 4;
+                        cardImgsURL.push(cardImgURL.substring(urlStartIndex));
+                    }
+                });
+                resolve(cardImgsURL)
+            })
+            .catch(function (err) {
+                // There was an error
+                console.warn("Something went wrong.", err);
```
Comment (wolfderechter) [inline packages/website/components/getImgURLs.ts:31]: also remove this

**Review by wolfderechter — CHANGES_REQUESTED** [review]

**Comment by Jaapiehetaapie** [comment]
Sorry for the mess. I'm not sure how to resolve the conflicts.

**Comment by dantaik** [comment]
The deployed preview doesn't show blogs fetched successfully! @Jaapiehetaapie

**Comment by dantaik** [comment]
Thank you for resolving the conflicts, @Jaapiehetaapie. I wonder if you can also display the blog entries as placeholders while the content is being fetched. Otherwise, during the loading period, that section looks wierd.

See https://jamesinkala.com/blog/make-animated-content-placeholders-with-html-and-css/

**Comment by Jaapiehetaapie** [comment]
I added the loading animation. The only problem there is right now is that i am using https://proxy.cors.sh/ to fetch the mirror site for the images. I'm am doing this to avoid CORS errors. I was wondering i someone knows a more robust fix for this?

**Comment by dantaik** [comment]
> I added the loading animation. The only problem there is right now is that i am using https://proxy.cors.sh/ to fetch the mirror site for the images. I'm am doing this to avoid CORS errors. I was wondering i someone knows a more robust fix for this? When fetching the post data from the arweave API i get an imgURI aswell, but i dont know where this points to, maybe that can help.

Does this article help?

https://dev.mirror.xyz/GjssNdA6XK7VYynkvwDem3KYwPACSU9nDWpR5rei3hw

**Comment by Jaapiehetaapie** [comment]
> > I added the loading animation. The only problem there is right now is that i am using https://proxy.cors.sh/ to fetch the mirror site for the images. I'm am doing this to avoid CORS errors. I was wondering i someone knows a more robust fix for this? When fetching the post data from the arweave API i get an imgURI aswell, but i dont know where this points to, maybe that can help.
> 
> 
> 
> Does this article help?
> 
> 
> 
> https://dev.mirror.xyz/GjssNdA6XK7VYynkvwDem3KYwPACSU9nDWpR5rei3hw

Yeah, I used this to get most of the post data. This data also includes a ImgURI but I don't know where that points to, or how to convert that imgURI to an url.

**Comment by Jaapiehetaapie** [comment]
Maybe the people that's place the post know where the imgURI points to.

**Comment by dantaik** [comment]
What's the URI look like?

The images on mirror.xyz are hosted at https://mirror-media.imgix.net, one example is:

https://mirror-media.imgix.net/publication-images/5Ed-TXJIB3LTC2HJdPuEN.png?height=512&width=1024&h=512&w=1024&auto=compress

**Comment by Jaapiehetaapie** [comment]
This is an example of the imgURI : QmeN9jeFWF4TnH6L9xCHPmvkodbWc83ex2yxvHzpvsfSmT

**Comment by d1onys1us** [comment]
one idea i'm thinking which could be simpler is like this (not sure if the arweave graphql API exposes this)
1. query for all of the posts made by the taiko labs mirror account on ARWeave (fetch), and get the content ID
2. for each of these, take the 3 most recent, then we can use the open graph metadata to render the image title and short description.

if this is simpler, i think we should do this instead and we can drop the reading time, author, etc. and the additional dependencies. whatever is exposed by the open graph metadata (like if you just share a mirror link), i think should be sufficient for this preview.

let me know if you need any help on it, or what you think of this idea.

**Comment by wolfderechter** [comment]
I see that you managed to get the images now, well done! Also nice getReadingTime solution

File: packages/website/styles.css [hunk 5]
```diff
@@ -1,3 +1,40 @@
...
+    position: relative;
+    height: 400px;
+    background-color: rgb(38 38 38);
+    z-index: 1;
+    margin-top: .5em;
+    border-radius: .5rem;
+    overflow: hidden;
+  }
+
+.activity{
```
Comment (d1onys1us) [inline packages/website/styles.css:15]: so, in general with the website, it is preferred to use tailwind wherever possible. i am a little bit of a CSS noob, could you describe to me what these are aiming to achieve? we could maybe move this into either 1) inline tailwind css or 2) create a component, which uses the inline tailwind css or css modules (re-usable). let me know if you have any questions about this, happy to help.

File: packages/website/package.json [hunk 6]
```diff
@@ -8,13 +8,21 @@
...
+    "jsdom": "^21.1.0",
+    "net": "^1.0.2",
     "next": "^13.1.4",
     "next-themes": "^0.2.1",
     "nextra": "^2.2.12",
     "nextra-theme-docs": "^2.2.12",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "sharp": "^0.31.3"
+    "sharp": "^0.31.3",
```
Comment (d1onys1us) [inline packages/website/package.json:23]: do we need all these dependencies and also the newly added "browser" section? i see that we are using the `arweave` dependency, but I wonder if we need these other ones like acorn, fs, jsdom, net, tls, unist-util-remove.

File: packages/website/components/getPosts.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+              `})
+    }).then((res) => res.json())
+      .then((response) => {
+        getPosts(response)
+      })
+      .catch();
+  }
+
+  async function getPosts(response) {
+    var posts = []
```
Comment (d1onys1us) [inline packages/website/components/getPosts.ts:54]: probably better to use `const` here, for both the `var` areas.

File: packages/website/components/getPosts.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+          if (data.hasOwnProperty('wnft')) {
+
+            // add the original digest
+            data["OriginalDigest"] = edge.node.tags[4].value;
+            posts.push(data)
+          }
+
+        }).catch();
+    }))
+    resolve(posts)
```
Comment (d1onys1us) [inline packages/website/components/getPosts.ts:70]: semicolon

File: packages/website/components/getPosts.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  }
+  getTransanctionIds()
```
Comment (d1onys1us) [inline packages/website/components/getPosts.ts:72]: semicolon

File: packages/website/styles.css (hunk 5 above)
Comment (Jaapiehetaapie) [reply packages/website/styles.css:15]: the load-wrapper and acticity class are resposible for the loading preview that shows while the posts are loading. The crop-image class is there to make sure only to show the upper half of the image. I'm not that good with css myself too and never worked with tailwind. But il will read up on tailwind and try to do it that way. If you have any suggestions please let me know.

**Review by dantaik — DISMISSED** [review]

**Comment by d1onys1us** [comment]
looks good, thanks @Jaapiehetaapie ! however I am not able to see the images on the preview deploy: https://website-k6ktlpx1o-taikoxyz.vercel.app/

image

do you know why?

**Comment by Jaapiehetaapie** [comment]
Weird. It's working for me, can you maybe check the logs to see if there are any erros when fetching from ipfs.

**Comment by d1onys1us** [comment]
> Weird. It's working for me, can you maybe check the logs to see if there are any erros when fetching from ipfs.

image

here is the error. it won't resolve in my browser either, if i just navigate to it in a separate tab. is there another indexer we can use? does https://ipfs.io/ipfs/QmeN9jeFWF4TnH6L9xCHPmvkodbWc83ex2yxvHzpvsfSmT resolve for you? if i open it up in IPFS itself yes I can see it, just can't retrieve it from here.

**Comment by wolfderechter** [comment]
[Website ](https://website-k6ktlpx1o-taikoxyz.vercel.app/) and [ipfs](https://ipfs.io/ipfs/QmeN9jeFWF4TnH6L9xCHPmvkodbWc83ex2yxvHzpvsfSmT) are working for me

**Comment by d1onys1us** [comment]
@Jaapiehetaapie there are some merge conflicts with package.json and pnpm-lock.yaml, i think because i updated the packages on main branch. could you please resolve them? if you need any help let me know. once again thank you so much for your effort!

**Review by dionysuzx — DISMISSED** [review]

**Comment by Jaapiehetaapie** [comment]
I don't know how to resolve the last conflict, because arweave is needed for the scripts to work

**Comment by Jaapiehetaapie** [comment]
I'm having problems resolving the conflict, does anyone know how to do this?

**Review by dionysuzx — APPROVED** [review]

**Comment by cyberhorsey** [comment]
@dantaik i think you are queued to review this, this PR is quite old now.

**Review by dantaik — APPROVED** [review]

**Pull request merged**